### PR TITLE
8214 k8s 1.27 upgrade

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description

Changes for k8s 1.27 upgrade

Most of the changes are in the generated documentation code, because we renamed the `master` branch to `main` a few months ago, and this is apparently the first time since then we've really regenerated all the code, and we see changes like
`##### Source File: [github.com/solo-io/gloo/projects/gloo/api/external/envoy/config/core/v3/http_uri.proto](https://github.com/solo-io/gloo/blob/**master**/projects/gloo/api/external/envoy/config/core/v3/http_uri.proto)`
to
`##### Source File: [github.com/solo-io/gloo/projects/gloo/api/external/envoy/config/core/v3/http_uri.proto](https://github.com/solo-io/gloo/blob/**main**/projects/gloo/api/external/envoy/config/core/v3/http_uri.proto)`
for example in [...ce/api/github.com/solo-io/gloo/projects/gateway/api/v1/http_gateway.proto.sk.md](https://github.com/solo-io/gloo/pull/8484/files#diff-ecb07e2c26e46962e27b5337c33650f2b542b63dea5b093ed7d9344dcb59aeac)

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/root.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func App(opts *options.Options, preRunFuncs []RunnableCommand, postRunFuncs []Ru
 
 		// If the given subcommand does not exist, look for a suitable plugin executable
 		if _, _, err := app.Find(cmdPathPieces); err != nil {
-			if err := cmd.HandlePluginCommand(pluginHandler, cmdPathPieces); err != nil {
+			if err := cmd.HandlePluginCommand(pluginHandler, cmdPathPieces, false); err != nil {
```
Comment (sheidkamp) [inline projects/gloo/cli/pkg/cmd/root.go:81]: This is a manual code update, the new arg is "exactMatch". Can't find release notes, but looking at the code, "false" retains current functionality:

Old code: https://github.com/kubernetes/kubectl/blob/release-1.26/pkg/cmd/cmd.go#L227
New code: https://github.com/kubernetes/kubectl/blob/release-1.27/pkg/cmd/cmd.go#L259

the false value gets us past this new `if`: https://github.com/kubernetes/kubectl/blob/13f7b15bc1b75003bb91ef2d2de63c84d4525e33/pkg/cmd/cmd.go#L279

**Review by elcasteel — DISMISSED** [review]

File: go.mod [hunk 2]
```diff
@@ -41,50 +41,50 @@ require (
...
-	helm.sh/helm/v3 v3.11.2
-	k8s.io/api v0.26.4
-	k8s.io/apiextensions-apiserver v0.26.4
-	k8s.io/apimachinery v0.26.4
-	k8s.io/client-go v0.26.4
-	k8s.io/code-generator v0.26.4
-	k8s.io/component-base v0.26.4
-	k8s.io/kubectl v0.26.0
-	k8s.io/utils v0.0.0-20221128185143-99ec85e7a448
+	helm.sh/helm/v3 v3.12.2
```
Comment (sam-heilbron) [inline go.mod:76]: We use helm 3.11 as the server version in our kube tests: https://github.com/solo-io/gloo/blob/main/.github/workflows/regression-tests.yaml#L48

Not critical, but I think it's good practice to keep these in sync. Can we update that version? (if we do that, we also have support docs that will need updating)

**Review by sam-heilbron — COMMENTED** [review]
If we're updating our kube library version, I think we should update the server version used in our regression tests as well: https://github.com/solo-io/gloo/blob/main/.github/workflows/regression-tests.yaml#L48

Also our nightly tests which validate the upper/lower k8s support window will need to be updated: https://github.com/solo-io/gloo/blob/main/.github/workflows/nightly-tests.yaml#L35

It may be worthwhile to make these changes and then run the nightly job manually from your branch to see if they will pass

File: go.mod (hunk 2 above)
Comment (sheidkamp) [reply go.mod:76]: From: the [Helm 3.12.0 release notes](https://github.com/helm/helm/releases/tag/v3.12.0):

`Bump k8s.io deps from v0.26.0 to v0.27.0 https://github.com/helm/helm/commit/b9361407a36e0f85ba7833b6b2804a21c1a409cc (Tom Wieczorek)`

and the builds fail with  Helm 3.11. What support docs need updating and how do we start that process?

File: changelog/v1.14.0-beta18/k8s-125.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ changelog:
   - type: DEPENDENCY_BUMP
     dependencyOwner: helm
     dependencyRepo: helm
-    dependencyTag: v3.11.2
+    dependencyTag: v3.12.2
```
Comment (sam-heilbron) [inline changelog/v1.14.0-beta18/k8s-125.yaml:21]: I think this was unintentional? I assume a find/replace

**Review by sam-heilbron — COMMENTED** [review]
In https://github.com/solo-io/gloo/blob/main/docs/content/reference/support.md we claim that we support helm 3.8. Can we update that to reflect these changes as well?

**Comment by sheidkamp** [comment]
> In https://github.com/solo-io/gloo/blob/main/docs/content/reference/support.md we claim that we support helm 3.8. Can we update that to reflect these changes as well?

Updated

File: test/mocks/kubernetes/kubeinterface.go [hunk 4]
```diff
@@ -37,22 +38,22 @@ import (
...
+	v1alpha15 "k8s.io/client-go/kubernetes/typed/node/v1alpha1"
 	v1beta111 "k8s.io/client-go/kubernetes/typed/node/v1beta1"
 	v112 "k8s.io/client-go/kubernetes/typed/policy/v1"
 	v1beta112 "k8s.io/client-go/kubernetes/typed/policy/v1beta1"
 	v113 "k8s.io/client-go/kubernetes/typed/rbac/v1"
-	v1alpha15 "k8s.io/client-go/kubernetes/typed/rbac/v1alpha1"
+	v1alpha16 "k8s.io/client-go/kubernetes/typed/rbac/v1alpha1"
 	v1beta113 "k8s.io/client-go/kubernetes/typed/rbac/v1beta1"
-	v1alpha16 "k8s.io/client-go/kubernetes/typed/resource/v1alpha1"
+	v1alpha2 "k8s.io/client-go/kubernetes/typed/resource/v1alpha2"
```
Comment (nfuden) [inline test/mocks/kubernetes/kubeinterface.go:56]: whats the difference in alpha 2

**Review by sam-heilbron — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Review by elcasteel — APPROVED** [review]

**Pull request merged**